submissions/merowing/js-post-oop/index.js

```diff
@@ -0,0 +1,94 @@
+import { print } from './js/lib.js';
+
+class Creature {
+   constructor(name, gender, species, hands) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.hands = hands;
+   }
+
+   message() {
+      return `${this.species}; <strong>${this.name}</strong>; ${this.gender}; ${this.hands};`;
+   }
+}
+
+class Human extends Creature {
+   constructor(name, gender, hands, legs, say, species) {
+      super(name, gender, species, hands);
+      this.say = say;
+      this.legs = legs;
+   }
+
+   set friends(names) {
+      this._friends = names.map(friend => friend.name);
+   }
+
+   get friends() {
+      return ` <strong>${this._friends}</strong>`;
+   }
+
+   message() {
+      return `${super.message()} ${this.legs}; <em>${this.say}</em>;${this.friends}`;
+   }
+
+}
+
+class Man extends Human {
+   constructor(name, gender, hands, legs, say, species = 'man') {
+      super(name, gender, hands, legs, say, species);
+   }
+}
+
+class Woman extends Human {
+   constructor(name, gender, hands, legs, say, species = 'woman') {
+      super(name, gender, hands, legs, say, species);
+   }
+}
+
+class FairyTaleCat extends Human {
+   constructor(name, gender, hands, legs, say = '', species = 'fairy tale cat') {
+      super(name, gender, hands, legs, say, species);
+   }
+}
+
+class Animal extends Creature {
+   constructor(name, gender, paws, say, species, hands = 0) {
+      super(name, gender, species, hands);
+      this.say = say;
```

Comment:
Why `legs` is generalized to the base class and `say` is not?
Not a rhetorical question.

Reply:
because, I don't know how to preserve output order.
Now, I return array from main class and split this array to get say string separately